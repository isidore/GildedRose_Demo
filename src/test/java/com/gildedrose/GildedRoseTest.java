package com.gildedrose;

import com.github.larseckart.tcr.FastTestCommitRevertMainExtension;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith(FastTestCommitRevertMainExtension.class)


public class GildedRoseTest {

    @Test
    public void foo() {
        String names[] = {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert","Sulfuras, Hand of Ragnaros"};
        Integer qualities[] = {0,-1,1};
        CombinationApprovals.verifyAllCombinations(this::doStuff, names, qualities);
    }

    private String doStuff(String name, Integer quality) {
        Item[] items = new Item[]{new Item(name, 0, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        String result = app.items[0].toString();
        return result;
    }

}
