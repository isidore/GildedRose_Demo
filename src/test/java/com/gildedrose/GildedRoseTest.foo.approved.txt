[foo, -1] => [foo, sellin: -1, quality: -1] 
[foo, 0] => [foo, sellin: -1, quality: 0] 
[foo, 1] => [foo, sellin: -1, quality: 0] 
[Aged Brie, -1] => [Aged Brie, sellin: -1, quality: 1] 
[Aged Brie, 0] => [Aged Brie, sellin: -1, quality: 2] 
[Aged Brie, 1] => [Aged Brie, sellin: -1, quality: 3] 
[Backstage passes to a TAFKAL80ETC concert, -1] => [Backstage passes to a TAFKAL80ETC concert, sellin: -1, quality: 0] 
[Backstage passes to a TAFKAL80ETC concert, 0] => [Backstage passes to a TAFKAL80ETC concert, sellin: -1, quality: 0] 
[Backstage passes to a TAFKAL80ETC concert, 1] => [Backstage passes to a TAFKAL80ETC concert, sellin: -1, quality: 0] 
[Sulfuras, Hand of Ragnaros, -1] => [Sulfuras, Hand of Ragnaros, sellin: 0, quality: -1] 
[Sulfuras, Hand of Ragnaros, 0] => [Sulfuras, Hand of Ragnaros, sellin: 0, quality: 0] 
[Sulfuras, Hand of Ragnaros, 1] => [Sulfuras, Hand of Ragnaros, sellin: 0, quality: 1] 
